src/shared/api/http-client.ts

```
@@ -0,0 +1,221 @@
+}
+
+export type RequestParams = Omit<FullRequestParams, 'body' | 'method' | 'query' | 'path'>;
+
+export interface ApiConfig<SecurityDataType = unknown> {
+  baseUrl?: string;
+  baseApiParams?: Omit<RequestParams, 'baseUrl' | 'cancelToken' | 'signal'>;
+  securityWorker?: (securityData: SecurityDataType | null) => Promise<RequestParams | void> | RequestParams | void;
+  customFetch?: typeof fetch;
+}
+
+export interface HttpResponse<D extends unknown, E extends unknown = unknown> extends Response {
```

Biome (1.9.4)

[error] Constraining a type parameter to any or unknown is useless.

All types are subtypes of any and unknown.
Safe fix: Remove the constraint.


(lint/complexity/noUselessTypeConstraint)